lib/kube/proxy/watcher.go

```diff
@@ -37,7 +37,16 @@ func (s *TLSServer) startReconciler(ctx context.Context) (err error) {
 		s.log.Debug("Not initializing Kube Cluster resource watcher.")
 		return nil
 	}
-	s.reconciler, err = services.NewReconciler(services.ReconcilerConfig{
+	s.reconciler, err = services.NewReconciler(services.ReconcilerConfig[types.KubeCluster]{
+		GetKind: func(cluster types.KubeCluster) string {
+			return cluster.GetKind()
+		},
+		GetName: func(cluster types.KubeCluster) string {
+			return cluster.GetName()
+		},
+		GetOrigin: func(cluster types.KubeCluster) string {
+			return cluster.Origin()
```

Comment:
If we constrained the Reconciler's [T] with an interface could we do away with this boilerplate?

Are we avoiding that because it wouldn't match 153 resources?